datasets/arabic_pos_dialect/arabic_pos_dialect.py

```diff
@@ -0,0 +1,129 @@
+
+import csv
+import datasets
+
+
+_CITATION = """\
+@InProceedings{DARWISH18.562,  author = {Kareem Darwish ,Hamdy Mubarak ,Ahmed Abdelali ,Mohamed Eldesouki ,Younes Samih ,Randah Alharbi ,Mohammed Attia ,Walid Magdy and Laura Kallmeyer},
+title = {Multi-Dialect Arabic POS Tagging: A CRF Approach},
+booktitle = {Proceedings of the Eleventh International Conference on Language Resources and Evaluation (LREC 2018)},
+year = {2018},
+month = {may},
+date = {7-12},
+location = {Miyazaki, Japan},
+editor = {Nicoletta Calzolari (Conference chair) and Khalid Choukri and Christopher Cieri and Thierry Declerck and Sara Goggi and Koiti Hasida and Hitoshi Isahara and Bente Maegaard and Joseph Mariani and Hélène Mazo and Asuncion Moreno and Jan Odijk and Stelios Piperidis and Takenobu Tokunaga},
+publisher = {European Language Resources Association (ELRA)},
+address = {Paris, France},
+isbn = {979-10-95546-00-9},
+language = {english}
+}
+"""
+
+_DESCRIPTION = """\
+The Dialectal Arabic Datasets contain four dialects of Arabic, Etyptian (EGY), Levantine (LEV), Gulf (GLF), and Maghrebi (MGR). Each dataset consists of a set of 350 manually segmented and POS tagged tweets.
+"""
+
+_URL = "https://github.com/qcri/dialectal_arabic_resources/raw/master/"
+_DIALECTS = ["egy", "lev", "glf", "mgr"]
+
+
+class ArabicPosDialectConfig(datasets.BuilderConfig):
+    """ BuilderConfig for ArabicPosDialect"""
+
+    def __init__(self, dialect=None, **kwargs):
+        """
+
+        Args:
+            dialect: the 3-letter code string of the dialect set that will be used.
+            Code should be one of the following: egy, lev, glf, mgr.
+            **kwargs: keyword arguments forwarded to super.
+        """
+        super(ArabicPosDialectConfig, self).__init__(**kwargs)
+        assert dialect in _DIALECTS, ("Invalid dialect code: %s", dialect)
+        self.dialect = dialect
+
+
+class ArabicPosDialect(datasets.GeneratorBasedBuilder):
+    """POS-tagged Arabic tweets in four major dialects."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        ArabicPosDialectConfig(
+            name=dialect,
+            dialect=dialect,
+            description="A set of 350 tweets in the {} dialect of Arabic that have been manually segmented and POS tagged.".format(
+                dialect
+            ),
+        )
+        for dialect in _DIALECTS
+    ]
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "Fold": datasets.Value("int32"),
+                    "SubFold": datasets.Value("string"),
+                    "SentID": datasets.Value("int32"),
+                    "Order": datasets.Value("int32"),
+                    "Word": datasets.Value("string"),
+                    "Segmentation": datasets.Value("string"),
+                    "POS": datasets.Value("string"),
```

Comment:
We can make the features more practical to use IMO if use the sentences instead of the words.
For example we can use these features:
```python
features=datasets.Features(
    {
        "Fold": datasets.Value("int32"),
        "SubFold": datasets.Value("string"),
        "tokens": datasets.Sequence(datasets.Value("string")),
        "ner_tags": datasets.Sequence(datasets.ClassLabel(names=ner_labels_list)),
        "pos_tags": datasets.Sequence(datasets.ClassLabel(names=pos_labels_list)),
    }
)
```
This way it's easier to use to train a model and it is also consistent with the format of the other NER and POS datasets

What do you think ?

Reply:
`Segmentation` doesn't refer to NER tags. It refers to morphemes. So I could see the sentence version being:
```
features=datasets.Features(
    {
        "Fold": datasets.Value("int32"),
        "SubFold": datasets.Value("string"),
        "tokens": datasets.Sequence(datasets.Value("string")),
        "segments": datasets.Sequence(datasets.Value("string")),
        "pos_tags": datasets.Sequence(datasets.ClassLabel(names=pos_labels_list)),
    }
)
```
The POS tags also correspond to the segments rather than the words, so a single word can have multiple POS tags connected with `+` like `PROG_PART+V+PRON`. We could collect all the POS tag combinations in the dataset to make the ClassLabel list, but I think decoupling the tags would result in a loss of information as you wouldn't have the word boundaries anymore.